src/components/node-list/node-list-tree.js

```diff
@@ -0,0 +1,207 @@
+// and setup is only used here and not anywhere else in the app.
+const useStyles = makeStyles({
+  root: {
+    height: 110,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+});
+
+const StyledTreeView = withStyles({
+  root: {
+    padding: '0 0 0 20px',
+  },
+})(TreeView);
+
+const isModularPipelineType = (type) => type === 'modularPipeline';
+
+const TreeListProvider = ({
+  nodes,
+  nodeSelected,
+  onToggleNodeSelected,
+  searchValue,
+  modularPipelines,
+  modularPipelineIds,
+  nodeModularPipelines,
+  onItemChange,
+  onItemMouseEnter,
+  onItemMouseLeave,
+  nodeTypeIDs,
+  searching,
+}) => {
+  const classes = useStyles();
+
+  const [expandedPipelines, setExpandedPipelines] = useState([]);
+
+  useEffect(() => {
+    const filteredTreeItems =
+      searchValue !== ''
+        ? getFilteredTreeItems({
+            nodes,
+            modularPipelines,
+            nodeSelected,
+            searchValue,
+            modularPipelineIds,
+            nodeModularPipelines,
+            nodeTypeIDs,
+          })
+        : [];
+
+    let expandedModularPipelines = [];
+
+    searchValue !== '' &&
+      filteredTreeItems.forEach((modularPipeline) =>
+        expandedModularPipelines.push(modularPipeline.id)
+      );
+    setExpandedPipelines(expandedModularPipelines);
+  }, [
+    searchValue,
+    nodes,
+    modularPipelines,
+    nodeSelected,
+    modularPipelineIds,
+    nodeModularPipelines,
+    nodeTypeIDs,
+  ]);
+
+  const treeData = getNestedModularPipelines({
+    nodes,
+    modularPipelines,
+    nodeSelected,
+    searchValue,
+    modularPipelineIds,
+    nodeModularPipelines,
+    nodeTypeIDs,
+  });
+
+  const onItemClick = (item) => {
+    if (!isModularPipelineType(item.type)) {
+      onToggleNodeSelected(item.id);
+    }
+  };
+
+  const renderModularPipelines = (treeData) =>
+    treeData.children.map((node) =>
+      renderTree(
+        node,
+        onItemMouseEnter,
+        onItemMouseLeave,
+        onItemChange,
+        onItemClick
+      )
+    );
+
+  const renderChildNodes = (treeData) =>
+    treeData.nodes.map((node) => (
+      <NodeListTreeItem
+        data={node}
+        onItemMouseEnter={onItemMouseEnter}
+        onItemMouseLeave={onItemMouseLeave}
+        onItemChange={onItemChange}
+        onItemClick={onItemClick}
+        key={node.id}
+      />
+    ));
+
+  const renderTree = (
+    rowData,
+    onItemMouseEnter,
+    onItemMouseLeave,
+    onItemChange,
+    onItemClick
+  ) => (
+    <NodeListTreeItem
+      data={rowData}
+      onItemMouseEnter={onItemMouseEnter}
+      onItemMouseLeave={onItemMouseLeave}
+      onItemChange={onItemChange}
+      onItemClick={onItemClick}
+      key={rowData.id}>
+      {renderModularPipelines(rowData)}
+
+      {/* render set of node elements in that modular pipeline */}
+      {renderChildNodes(rowData)}
+    </NodeListTreeItem>
+  );
+
+  return searching ? (
```

Comment:
These look roughly the same, can just one `StyledTreeView` be used here and then conditionals on the attributes instead?

Just thought worth noting here too that I'd say usually prefer to go for an `if` statement block first over using the ternary `?` where it's not a short and trivial assignment.

Reply:
haha yea in respond to this - please see my previous answer to Rashida's question. The main reason we need to specify a seperate `StyledTreeView` is the need to reload that in going between the normal tree-view and the search Tree, which is an expanded tree. ( note the need to specify the `expanded` prop for the searching condition). Passing a new value in between the searching and the non-search scenerio would cause a react error on the browser as we are forcing to switch between a controlled and uncontrolled react component.) Doing so would force the component to remount and allow stability in performance, as well as to elinate the react error.

Reply:
Ah okay if it's a limitation of the framework, I guess worth adding that in a comment here?